### ISSUE TITLE
Cryptic segfault in pytorch2.2

### ISSUE THREAD

**Issue by arogozhnikov** [pytorch/pytorch#122385]
### 🐛 Describe the bug

Following code segfaults in pytorch 2.2. It works ok with previous version of pytorch (2.1)

```python
import torch

def _kabsch(P, Q):
    u, _, vt = torch.linalg.svd(torch.matmul(P.T, Q).to(torch.float32))
    s = torch.eye(P.shape[1], device=P.device)
    s[-1, -1] = torch.sign(torch.linalg.det(torch.matmul(u, vt))) # segfault happens here
    r_opt = torch.matmul(torch.matmul(u, s), vt)
    return r_opt.to(device=P.device, dtype=P.dtype)

device = torch.device('cuda:0')
_kabsch(
    torch.randn(512, 3).to(device),
    torch.randn(512, 3).to(device),
)
```

### Versions

```
Collecting environment information...
PyTorch version: 2.2.1+cu121
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.6 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.31

Python version: 3.10.13 | packaged by conda-forge | (main, Dec 23 2023, 15:36:39) [GCC 12.3.0] (64-bit runtime)
Python platform: Linux-5.4.0-171-generic-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: 12.2.140
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: NVIDIA A100-SXM4-80GB
GPU 1: NVIDIA A100-SXM4-80GB
GPU 2: NVIDIA A100-SXM4-80GB
GPU 3: NVIDIA A100-SXM4-80GB
GPU 4: NVIDIA A100-SXM4-80GB
GPU 5: NVIDIA A100-SXM4-80GB
GPU 6: NVIDIA A100-SXM4-80GB
GPU 7: NVIDIA A100-SXM4-80GB

Nvidia driver version: 535.154.05
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Byte Order:                         Little Endian
Address sizes:                      46 bits physical, 57 bits virtual
CPU(s):                             96
On-line CPU(s) list:                0-95
Thread(s) per core:                 2
Core(s) per socket:                 24
Socket(s):                          2
NUMA node(s):                       2
Vendor ID:                          GenuineIntel
CPU family:                         6
Model:                              106
Model name:                         Intel(R) Xeon(R) Gold 6338 CPU @ 2.00GHz
Stepping:                           6
CPU MHz:                            2000.042
BogoMIPS:                           4000.08
Virtualization:                     VT-x
Hypervisor vendor:                  KVM
Virtualization type:                full
L1d cache:                          1.5 MiB
L1i cache:                          1.5 MiB
L2 cache:                           192 MiB
L3 cache:                           32 MiB
NUMA node0 CPU(s):                  0-47
NUMA node1 CPU(s):                  48-95
Vulnerability Gather data sampling: Unknown: Dependent on hypervisor status
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown
Vulnerability Retbleed:             Not affected
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Mitigation; TSX disabled
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon rep_good nopl xtopology cpuid tsc_known_freq pni pclmulqdq vmx ssse3 fma cx16 pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch cpuid_fault invpcid_single ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves wbnoinvd arat avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq rdpid md_clear arch_capabilities

Versions of relevant libraries:
[pip3] mypy==1.8.0
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.26.4
[pip3] pytorch-lightning==2.2.1
[pip3] torch==2.2.1
[pip3] torchmetrics==1.3.1
[pip3] triton==2.2.0
[conda] numpy                     1.26.4                   pypi_0    pypi
[conda] pytorch-lightning         2.2.1                    pypi_0    pypi
[conda] torch                     2.2.1                    pypi_0    pypi
[conda] torchmetrics              1.3.1                    pypi_0    pypi
[conda] triton                    2.2.0                    pypi_0    pypi
```

cc @ptrblck

**Comment by arogozhnikov** [comment]
Segfault happens in `linalg.det`. 
Specific set of conditions is unclear: just passing 3x3 matrix works, passing a matrix which is a product of 3x3 matrices also works. So problem is likely in previous operations 🤷

**Comment by arogozhnikov** [comment]
ah wait, that's actually simpler:
```python
import torch
device = torch.device('cuda:0')
x = torch.randn(3, 3)
torch.linalg.det(x.to(device)) # segfaults
```

**Comment by KungYork** [comment]
Hello, it seems OK on torch-2.3.0. Could you have a try on torch-2.3.0?

**Comment by arogozhnikov** [comment]
@KungYork  I can confirm that on same hardware and  --pre 2.3.0.* this problem is not observed. 

But will not use non-stable torch for work.

**Comment by arogozhnikov** [comment]
this is resolved in torch 2.3
